app/Services/Post/PostService.php

```diff
@@ -58,7 +74,24 @@ public function edit($id)
      */
     public function update($request, $id)
     {
-        return $this->postDao->update($request, $id);
+        $image = $request->file('file');
+        if (is_null($image)) {
+            $imageName = 'default.png';
+        } else {
+            $imageName = $image->getClientOriginalName();
+            $location = 'images';
+            $image->move($location, $imageName);
+        }
+
+        $data = (object) [
+            'title' => $request->title,
+            'description' => $request->description,
+            'status' => $request->status,
+            'categories' => $request->categories,
+            'image' => $imageName
+        ];
+
+        return $this->postDao->update($data, $id);
```

Comment:
**change** 
```
$image = $request->file('file');
$imageName = null;
if (!is_null($image)) {
    $imageName = $image->getClientOriginalName();
    $location = 'images';
    $image->move($location, $imageName);
}
```

**Remove**
```
 $data = (object) [
            'title' => $request->title,
            'description' => $request->description,
            'status' => $request->status,
            'categories' => $request->categories,
            'image' => $imageName
        ];
```

Add

`$request->image = $imageName;`
`return $this->postDao->update($request, $id);`